components/script/dom/htmllinkelement.rs

```diff
@@ -79,11 +79,16 @@ impl HTMLLinkElement {
     }
 
     pub fn set_stylesheet(&self, s: Arc<Stylesheet>) {
-        assert!(self.stylesheet.borrow().is_none());
-        *self.stylesheet.borrow_mut() = Some(s);
+        let mut sheet = self.stylesheet.borrow_mut();
+        let must_invalidate = sheet.is_some();
```

Comment:
It might be better to just leave the assert there as it is and do what @jdm suggested, which would fix the underlying issue instead of clear the symptoms. That assert will continue to be useful to find timing issues such as what caused this issue in the first place.